### PR TITLE
Add endpoint to send email notification

### LINKED ISSUE #83: Allow the user to provide an email address from the recommendation page
If the user did not provide an email address at the beginning user does not get any notification about the recommendations. 

In order to send a notification to the user with a link to the recommendations we should allow the user to provide an email address on the recommendations page.

### PR DESCRIPTION
**Pull request by bumi**
this also allows the user to set the email address

untested PR for @lewisdylan 

Closes #83

### CODE REVIEW

File: app/controllers/api/iterations_controller.rb [hunk 1]
```diff
@@ -22,5 +22,20 @@ def create
         render json: @iteration.errors, status: :unprocessable_entity
       end
     end
+
+    def notify_user
+      if current_user.update(user_params)
+        current_iteration.notify_completion
```
Comment (lewisdylan) [inline app/controllers/api/iterations_controller.rb]: Shouldn't we first update the user with the new email address and then notify the user

File: app/javascript/component/form/UserEmail.vue [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    export default {
+        name: "UserEmail",
+        data(){
+            return {
+                form: new Form({
+                    email:''
+                })
+            }
+        }, methods:{
+            save(){
```
Comment (bumi) [inline app/javascript/component/form/UserEmail.vue]: we somehow need the iteration id in here. do we have that? 
that form needs to be send to the iterations controller.

File: app/controllers/api/iterations_controller.rb (hunk 1 above)
Comment (bumi) [reply app/controllers/api/iterations_controller.rb]: the line 27 is updating the user. 
so if that was successful then in line 28 we can send the notification

Comment (lewisdylan) [reply app/controllers/api/iterations_controller.rb]: Sorry my bad, I didn't see it

File: app/javascript/component/form/UserEmail.vue [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      }),
+      completed: false,
+    }
+  },
+  methods: {
+    save() {
+      this.form.put(Iteration.url(this.iterationId, '/notify_user')).then(({ data }) => {
+        this.form.reset();
+        this.completed = true;
+        console.log(data);
```
Comment (bumi) [inline app/javascript/component/form/UserEmail.vue]: We should remove the debug outputs `console.log`

and I was thinking about the wording of `completed` - I felt like a form is `submitted`  - but just a thought.

**Pull request merged**